pkg/recovery/repair.go

```diff
@@ -64,18 +91,40 @@ func NewRepairHandler(s storage.Storer, logger logging.Logger, pushSyncer pushsy
 		}
 
 		// retrieve the chunk from the local store
-		ch, err := s.Get(ctx, storage.ModeGetRequest, swarm.NewAddress(chAddr))
+		ch, err := s.Get(ctx, storage.ModeGetRequest, addrToRetrieve)
 		if err != nil {
-			logger.Tracef("chunk repair: error while getting chunk for repairing: %v", err)
 			return err
 		}
 
 		// push the chunk using push sync so that it reaches it destination in network
 		_, err = pushSyncer.PushChunkToClosest(ctx, ch)
 		if err != nil {
-			logger.Tracef("chunk repair: error while sending chunk or receiving receipt: %v", err)
+			return err
+		}
+
+		// construct a new soc chunk from the soc envelope by adding the chunk data to the envelope
+		recoveryResponse := newRecoveryResponseChunk(envelopeAddress, m.Payload, ch.Data())
+
+		// send the soc envelope which has the chunk data, to the requester
+		_, err = pushSyncer.PushChunkToClosest(ctx, recoveryResponse)
+		if err != nil {
 			return err
 		}
 		return nil
```

Comment:
```suggestion
		return err
```

Reply:
why is lint not catching this... is code simplification not enabled in our lint?